idaes/core/util/misc.py

```diff
@@ -213,18 +213,25 @@ def svg_tag(
 
 
 # Author: John Eslick
-def copy_port_values(destination, source):
+def copy_port_values(destination=None, source=None, arc=None):
     """
     Copy the variable values in the source port to the destination port. The
     ports must containt the same variables.
 
     Args:
-        (pyomo.Port): Copy values from this port
-        (pyomo.Port): Copy values to this port
+        destination (Port): Copy values from this port
+        source (Port): Copy values to this port
+        arc (Arc): If arc is provided, use it to get default values of source
+            and destination
 
     Returns:
         None
     """
+    if arc is not None:
+        if source is None:
```

Comment:
It seems a bit odd to me to allow a user to specify a source/destination along with an Arc. For one, if a user were to specify all three then the `arc` argument is effectively ignored.

This won't stop the method from working, but it might be confusing to the user. Notably, the doc string says that the `arc` is used to define source and destination if provided, which is not quite correct.

I would be inclined to be restrictive here and make it an either/or approach - i.e. raise an Exception if a user provides both an `arc` and a `source` or `destination`.